src/background/network.js

```diff
@@ -10,9 +10,42 @@ const DOH_REQUEST_TIMEOUT = 30000; // 30 secs
 // Timeout between 1 network error and the next one.
 const NET_ERROR_TIMEOUT = 5000; // 5 seconds.
 
+const NET_ERROR_WARNING_TIMEOUT = 2000 // 2 seconds.
+
 // Telemetry host
 const TELEMETRY_HOST = "https://incoming.telemetry.mozilla.org";
 
+// Simple errors counter. If in NET_ERROR_WARNING_TIMEOUT milliseconds, we do
+// not receive successully connections, and the proxy is on, we try to increase
+// the connection isolation to recover from a potential network failure.
+class NetworkErrorCounter {
+  constructor(network) {
+    this.network = network;
+    this.errors = 0;
+    this.timerId = 0;
+  }
+
+  onError(error) {
+    if (this.errors === 0) {
```

Comment:
can this be `if(this.errors++ === 0) {` and kill the `++this.errors;` line below or is that confusing?